core/src/main/java/org/elasticsearch/action/ActionListenerResponseHandler.java

```diff
@@ -36,6 +38,20 @@ public ActionListenerResponseHandler(ActionListener<Response> listener) {
         this.listener = listener;
     }
 
+    /**
+     * Convenience method for delegating a response provided by supplier to the provided transport channel
+     */
+    public static <T extends TransportResponse> ActionListenerResponseHandler responseHandler(ActionListener<T> listener, Supplier<T>
```

Comment:
Nice guy that I am, I did that for you [here](https://github.com/elastic/elasticsearch/pull/17752) :smile: